From d97dec95a997033089dbd4d4ab38b30323c7fba3 Mon Sep 17 00:00:00 2001
From: Tony Lin <tony.lin@freescale.com>
Date: Thu, 11 Aug 2011 17:00:44 +0800
Subject: [PATCH 0340/2463] ENGR00139261-1 [esdhc]add 8 bit mode support para
 in platform data

set to 1 if the port on board supports 8 bit MMC card.
else set to 0

Signed-off-by: Tony Lin <tony.lin@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |    2 ++
 arch/arm/plat-mxc/include/mach/esdhc.h   |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index 21ae399..80d7695 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -241,11 +241,13 @@ static const struct esdhc_platform_data mx6q_sabreauto_sd3_data __initconst = {
 	.cd_gpio = MX6Q_SABREAUTO_SD3_CD,
 	.wp_gpio = MX6Q_SABREAUTO_SD3_WP,
 	.support_18v = 1,
+	.support_8bit = 1,
 };
 
 /* No card detect signal for SD4 */
 static const struct esdhc_platform_data mx6q_sabreauto_sd4_data __initconst = {
 	.always_present = 1,
+	.support_8bit = 1,
 };
 
 /* The GPMI is conflicted with SD3, so init this in the driver. */
diff --git a/arch/arm/plat-mxc/include/mach/esdhc.h b/arch/arm/plat-mxc/include/mach/esdhc.h
index de64035..cb34cd8 100644
--- a/arch/arm/plat-mxc/include/mach/esdhc.h
+++ b/arch/arm/plat-mxc/include/mach/esdhc.h
@@ -24,5 +24,6 @@ struct esdhc_platform_data {
 	unsigned int cd_gpio;
 	unsigned int always_present;
 	unsigned int support_18v;
+	unsigned int support_8bit;
 };
 #endif /* __ASM_ARCH_IMX_ESDHC_H */
-- 
1.7.7.4

